### PR TITLE
Fixing HIP Algorithm Configuration V2

### PR DESCRIPTION
**Pull request by mmusich**
Fixing track-refitting sequences to avoid MeasurementTrackerEvent issues as discussed in other PRs such as #6244 and #6233. Cleanup of the Cosmic data selection and introduction of the exception when using starting conditions for the Global Position Record with multiple IOVs as in V00-28-00 cvs tag (never entered in release). To be fixed in a second time: how to get highest possible run number for the starting conditions check

### CODE REVIEW

**Comment by mmusich** [comment]
@cmsbuild please test

**Comment by diguida** [comment]
It seems a problem with RECO geometry.

````
----- Begin Fatal Exception 06-Feb-2015 19:57:37 CET-----------------------
An exception of category 'DDException' occurred while
   [0] Processing run: 1 lumi: 4 event: 301
   [1] Running path 'p'
   [2] Calling event method for module PFDisplacedVertexProducer/'particleFlowDisplacedVertex'
   [3] Using EventSetup component GlobalTrackingGeometryESProducer/'' to make data GlobalTrackingGeometry/'' in record GlobalTrackingGeometryRecord
   [4] Using EventSetup component TrackerDigiGeometryESModule/'trackerGeometry' to make data TrackerGeometry/'' in record TrackerDigiGeometryRecord
   [5] Using EventSetup component TrackerGeometricDetESModule/'trackerNumberingGeometry' to make data GeometricDet/'' in record IdealGeometryRecord
Exception Message:
DDVectorGetter: vectors are empty, cannot get array detIdShifts
````

@ianna can you please have a look? It is not a `boost` serialisation issue, as the payload was exported with 73X, as you can see from the insertion time [here](https://cms-conddb.cern.ch/browser/#list/Prod/tags/TKRECO_Geometry_2015_71YV3), and a GT test loads the payload without problems.

**Comment by mmusich** [comment]
@cmsbuild please test

File: Alignment/HIPAlignmentAlgorithm/src/HIPAlignmentAlgorithm.cc [hunk 1]
```diff
@@ -98,6 +104,19 @@ HIPAlignmentAlgorithm::initialize( const edm::EventSetup& setup,
 				   AlignmentParameterStore* store )
 {
   edm::LogWarning("Alignment") << "[HIPAlignmentAlgorithm] Initializing...";
+
+  edm::ESHandle<Alignments> globalPositionRcd;
+  // FIXME! temporary solution to get highest possible run number
```
Comment (mmusich) [inline Alignment/HIPAlignmentAlgorithm/src/HIPAlignmentAlgorithm.cc]: more refined solution will come in later fix

**Comment by mmusich** [comment]
+1
mandatory changes to allow HIP alignment algorithm to run in releases past 53X

**Comment by davidlange6** [comment]
+1

**Pull request merged**